backend/src/main/java/de/otto/platform/gitactionboard/config/security/WebSecurityConfig.java

```diff
@@ -127,6 +131,27 @@ public List<UserDetails> getBasicAuthUsers(
           .collect(Collectors.toList());
     }
 
+    @Bean(name = "basicAuthUsers")
+    @ConditionalOnExpression(
+        "T(org.springframework.util.StringUtils).hasText('${BASIC_AUTH_USER_DETAILS:}')")
+    public List<UserDetails> getBasicAuthUsersFromContent(
+        @Value("${BASIC_AUTH_USER_DETAILS}") String basicAuthFileContent) throws IOException {
+      return Arrays.stream(basicAuthFileContent.split(System.lineSeparator()))
+          .filter(line -> !line.isBlank())
+          .map(
+              line -> {
+                final String[] credentials = line.split(CREDENTIAL_SEPARATOR);
+                return new AbstractMap.SimpleImmutableEntry<>(credentials[0], credentials[1]);
+              })
+          .map(
+              authDetails ->
+                  User.withUsername(authDetails.getKey())
+                      .password(authDetails.getValue())
+                      .authorities("ROLE_USER")
+                      .build())
+          .collect(Collectors.toList());
```

Comment:
This is a duplicate code available on the same file. Please refactor it.

It'll be easy to refactor once you take care of the comment mentioned for backend/src/main/java/de/otto/platform/gitactionboard/config/GithubActionConfig.java